File: 206-reverse-linked-list/206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        # Navigate through the linked list and add each nodes value to an array
        array = []
        list1 = head
        while list1:
            array.append(list1.val)
            list1 = list1.next
        # reverse the array
        array.reverse()
        # Add the values to a linked list and return the linked list
        list2 = head
        for item in array:
            list2.val = item
            list2 = list2.next
        
        return head

File: 206-reverse-linked-list/206-reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    const items = [];
    //Navigate through the linked list and add each node value to an array
    let list = head;
    
    while (list){
        items.push(list.val);
        list = list.next;
    };
    //Reverse the array
    items.reverse();
    //Add the values to a linked list
    
    let list2 = head;
    
    for (let item of items){
        list2.val = item;
        list2 = list2.next;
    }
    
    return head;
    
};